src/Core/Checkout/Cart/Subscriber/BeforeCartSaved/LineItemsBeforeCartSavedSubscriber.php

```diff
@@ -0,0 +1,25 @@
+<?php declare(strict_types=1);
```

Comment:
using multiple event subscribers as this allows the developer to disable / decorate only one of those and not require all of them to be disabled / adjusted.